.github/workflows/build.yaml

```
@@ -13,6 +13,35 @@ permissions:
   contents: read
 
 jobs:
+  clang-format-check:
+    name: Clang Format Check
+    runs-on: macos-15
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          fetch-depth: 1
+
+      - name: Install Clang Format
+        run: brew install clang-format
+
+      - name: Run Clang Format Check
+        run: |
+          # Format all source - fi the source is formatted properly this will do nothing
+          clang-format *.c *.h client/*.c
```

Needs "in-place" option to update the files; otherwise it just writes to stdout.

```suggestion
          clang-format -i *.c *.h client/*.c
```

---

If we `-i` It will directly changes file if this is needed I will do.

---

It is needed; otherwise `git diff` will always succeed because nothing has changed.